### PR TITLE
[WM-2263] Update workflow output autofill behavior

### PR DESCRIPTION
**Pull request by dillydally414**
### What
- Updates to the output autofill behavior in Azure workflows to not autofill every empty output by default

### Why
- If a user is configuring a workflow with a template (e.g. from one of our featured COVID-19 workflows) we should not autofill empty outputs, and should only provide the output values from the template
- If a user is configuring a workflow that they have run before we should not autofill empty outputs, and should only provide the output values from the previous run (this would also apply to saved configs in the future)
- If a user is configuring a workflow that has not been run before and does not have a template (e.g. they just imported a new workflow from Github), all outputs should be auto-filled by default

### Testing strategy
- Unit tests
- Tested in UI

### CODE REVIEW

**Review by salonishah11 — APPROVED** [review]

File: src/workflows-app/SubmissionConfig.js [hunk 1]
```diff
@@ -118,12 +118,17 @@ export const BaseSubmissionConfig = (
...
-            _.get('destination.type', outputDef) === 'none'
-              ? _.set('destination', { type: 'record_update', record_attribute: parseMethodString(outputDef.output_name).variable || '' })(outputDef)
-              : outputDef
-          )(maybeParseJSON(newRunSetData.output_definition))
+          shouldUseOutputs
+            ? jsonParsedOutput
+            : jsonParsedOutput.map((outputDef) => ({
+                ...outputDef,
+                destination: { type: 'record_update', record_attribute: parseMethodString(outputDef.output_name).variable || '' },
+              }))
```
Comment (cjllanwarne) [inline src/workflows-app/SubmissionConfig.js]: TOL how would you feel about extracting this into a separate method so we could unit test the extract/analysis functionality separately from the page layout? We're kind of doing both in one test below, but that makes it hard to test edge cases (and since it's buried so deep, see exactly how this function is supposed to be working)

**Review by cjllanwarne — APPROVED** [review]
Looks good, one thought about a small refactor to enable more (short, functional) unit tests
